File: src/main/java/net/jgp/books/spark/ch17/lab220_analytics_author/MeetingsPerOrganizerApp.java
package net.jgp.books.spark.ch17.lab220_analytics_author;

import static org.apache.spark.sql.functions.col;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class MeetingsPerOrganizerApp {

  public static void main(String[] args) {
    MeetingsPerOrganizerApp app =
        new MeetingsPerOrganizerApp();
    app.start();
  }

  /**
   * The processing code.
   */
  private void start() {
    // Creates a session on a local master
    SparkSession spark = SparkSession.builder()
        .appName("JSON Lines to Dataframe")
        .master("local[*]")
        .getOrCreate();

    Dataset<Row> df = spark.read().format("delta")
        .load("/tmp/delta_grand_debat_events");

    df = df.groupBy(col("authorType")).count()
        .orderBy(col("authorType").asc_nulls_last());

    df.show(25, 0, false);
    df.printSchema();
  }
}

File: src/main/python/lab220_analytics_author/meetingsPerOrganizerApp.py
"""
 Author/Organizer analytics.

 @author rambabu.posa
"""
from pyspark.sql import (SparkSession, functions as F)

def main(spark):
    df = spark.read.format("delta") \
        .load("/tmp/delta_grand_debat_events")

    df = df.groupBy(F.col("authorType")) \
        .count() \
        .orderBy(F.col("authorType").asc_nulls_last())

    df.show(25, 0, False)
    df.printSchema()

if __name__ == "__main__":
    # Creates a session on a local master
    # To use Databricks Delta Lake, we should add delta core packages to SparkSession
    spark = SparkSession.builder \
        .appName("JSON Lines to Dataframe") \
        .config("spark.jars.packages", "io.delta:delta-core_2.12:0.7.0") \
        .master("local[*]").getOrCreate()

    # setting log level, update this as per your requirement
    spark.sparkContext.setLogLevel("warn")

    main(spark)
    spark.stop()